crates/core/machine/src/syscall/precompiles/u256x2048_mul/air.rs

```diff
@@ -0,0 +1,440 @@
+                        );
+                        cols.hi_ptr_memory.populate(
+                            event.channel,
+                            event.hi_ptr_memory,
+                            &mut new_byte_lookup_events,
+                        );
+
+                        //  Populate memory columns.
+                        for i in 0..WORDS_FIELD_ELEMENT {
+                            cols.a_memory[i].populate(
+                                event.channel,
+                                event.a_memory_records[i],
+                                &mut new_byte_lookup_events,
+                            );
+                        }
+
+                        for i in 0..WORDS_FIELD_ELEMENT * 8 {
+                            cols.b_memory[i].populate(
+                                event.channel,
+                                event.b_memory_records[i],
+                                &mut new_byte_lookup_events,
+                            );
+                        }
+
+                        for i in 0..WORDS_FIELD_ELEMENT * 8 {
+                            cols.lo_memory[i].populate(
+                                event.channel,
+                                event.lo_memory_records[i],
+                                &mut new_byte_lookup_events,
+                            );
+                        }
+
+                        for i in 0..WORDS_FIELD_ELEMENT {
+                            cols.hi_memory[i].populate(
+                                event.channel,
+                                event.hi_memory_records[i],
+                                &mut new_byte_lookup_events,
+                            );
+                        }
+
+                        let a = BigUint::from_bytes_le(&words_to_bytes_le::<32>(&event.a));
+                        let b_array: [BigUint; 8] = event
+                            .b
+                            .chunks(8)
+                            .map(|chunk| BigUint::from_bytes_le(&words_to_bytes_le::<32>(chunk)))
+                            .collect::<Vec<_>>()
+                            .try_into()
+                            .unwrap();
+
+                        let effective_modulus = BigUint::one() << 256;
+
+                        let mut carries = vec![BigUint::zero(); 9];
+                        let mut ab_plus_carry_cols = [
+                            &mut cols.a_mul_b1,
+                            &mut cols.ab2_plus_carry,
+                            &mut cols.ab3_plus_carry,
+                            &mut cols.ab4_plus_carry,
+                            &mut cols.ab5_plus_carry,
+                            &mut cols.ab6_plus_carry,
+                            &mut cols.ab7_plus_carry,
+                            &mut cols.ab8_plus_carry,
+                        ];
+
+                        for (i, col) in ab_plus_carry_cols.iter_mut().enumerate() {
+                            let (_, carry) = col.populate_mul_and_carry(
```

Comment:
I don't think you should have a special populate function.  E.g. it should route through the `populate` function.